### DISCUSSION TITLE
Install without Docker

### DISCUSSION THREAD

**Discussion by mikeamcbrien** [seatplus/seatplus#6] [installation]
I am looking for ways to install SeatPlus OUTSIDE of Docker. Do you have any tips or start of documentation to do this? Additionally I am willing to create the documentation for those of us that prefer not to use docker!

I was working way through the process and there seems to be many things that might be missed when not using Docker.

1. The Artisan Migrate command for core fails on a requirments that looks to be pulled from the eveapi install?
`#6 /Users/redeacted/repos/seatplus/core/vendor/seatplus/eveapi/database/migrations/2020_10_06_193916_add_status_and_comment_to_applications_table.php(44): Illuminate\\Support\\Facades\\Facade::__callStatic('table', Array)` specifically the artisan migrate shows ` SQLSTATE[42000]: Syntax error or access violation: 1101 BLOB, TEXT, GEOMETRY or JSON column 'comment' can't have a default value1
2. I found that when connecting a character, the 'token' column in the 'refresh_tokens' table was limited to 255, I updated this 10 1024 before it would work, indicating the token is too long for the column.
3. I found that scopes required a value and ran `ALTER TABLE `refresh_tokens` MODIFY `scopes` VARCHAR(255) NULL;` to allow the query to continue.

Thank you for this project, i'd love to get over these hurdles, get my local setup, and contribute more!

**Comment by herpaderpaldent** [comment]
Hi @mikeamcbrien 

Thank you for taking your time and reporting this so carefully. Before i try to help you more in detail, could you elaborate what you have done exactly? Especially these questions are interesting for me:

1. What DBMS do you use? What version?
2. When you installed `seatplus/core`, how did you install it or better: are you positive, that you are using the 4.x branch?

Regarding your questions:

1. i can't really tell without knowing the first two questions of mine
2. from the looks of it, not all migrations run at your end. Referring to this migration (https://github.com/seatplus/eveapi/blob/d7d2c4b9eecf9596a05f92c03a5ea3bb7f7ea480/database/migrations/2021_09_27_141359_update_refresh_tokens_table_to_support_v2_tokens.php#L13)  the coulm should have a maximum length of 65535 characters
3. again it looks like not all migrations ran through: the same migration dropped the scopes column from the table

---

Edit: If you need a little guidline on how to install it you might digest it from the https://github.com/seatplus/base-app/ repo.

i would always start here:
```shell
composer create-project seatplus/core . --prefer-dist --no-dev --no-ansi`
```

afterwards you need to edit the `.env` by adding the stuff that you would add through the bootstrap shell script. 
Finally you should run migrations, install and build the js app and this should pretty much be it.

When you get it to install would you leave a comment here how you have done it for others to follow?

**Comment by mikeamcbrien** [comment]
Perfect thanks for setting up a discussion...

My local install is using Valet, PHP 8.1, MySql v 8.0.27 through DBNgin, I installed Redis locally as well. I deleted everything and documented each step to my first error.

1. I cloned v4.0 from this repo - https://github.com/seatplus/core using `git clone https://github.com/seatplus/core .`
2. I ran `Composer Update`
3. Updated/added my .env file (see below)
4. Setup local database to match .env
5. ran `php artisan migrate` and that's when I get my first error....

```php artisan migrate

   INFO  Preparing database.  

  Creating migration table ............................................................................................................... 32ms DONE

   INFO  Running migrations.  

  2014_10_12_100000_create_password_reset_tokens_table .................................................................................... 9ms DONE
  2019_04_29_153658_create_global_settings_table .......................................................................................... 6ms DONE
  2019_05_01_115854_create_users_table .................................................................................................... 6ms DONE
  2019_05_01_120000_create_character_infos_table .......................................................................................... 8ms DONE
  2019_05_01_124934_create_refresh_tokens_table ........................................................................................... 7ms DONE
  2019_05_01_135544_create_character_users_table ......................................................................................... 18ms DONE
  2019_05_30_062428_create_failed_jobs_table .............................................................................................. 3ms DONE
  2019_09_14_053230_create_permission_tables ............................................................................................. 66ms DONE
  2019_09_14_120453_create_affiliations_table ............................................................................................. 3ms DONE
  2019_09_18_185641_create_corporation_infosTable ......................................................................................... 3ms DONE
  2019_10_26_162816_create_alliance_infos_table ........................................................................................... 7ms DONE
  2019_11_09_091613_create_character_roles_table ......................................................................................... 15ms DONE
  2019_11_20_130209_create_character_assets_table ......................................................................................... 7ms DONE
  2019_12_03_155900_create_universe_names_table ........................................................................................... 6ms DONE
  2019_12_08_180242_create_universe_types_table ........................................................................................... 6ms DONE
  2019_12_09_133742_create_universe_groups_table .......................................................................................... 7ms DONE
  2019_12_09_153542_create_universe_categories_table ...................................................................................... 6ms DONE
  2019_12_15_112342_create_character_affiliations_table ................................................................................... 6ms DONE
  2019_12_22_092342_create_universe_locations_table ....................................................................................... 7ms DONE
  2019_12_24_072442_create_universe_structures_table ...................................................................................... 7ms DONE
  2019_12_24_093142_create_universe_stations_table ........................................................................................ 7ms DONE
  2020_01_13_214342_create_universe_systems_table ......................................................................................... 6ms DONE
  2020_01_14_100742_create_universe_constellations_table .................................................................................. 8ms DONE
  2020_01_14_100742_create_universe_regions_table ......................................................................................... 6ms DONE
  2020_02_20_132142_create_sso_scopes_table ............................................................................................... 6ms DONE
  2020_02_26_134742_create_applications_table ............................................................................................. 2ms DONE
  2020_04_02_191442_create_schedules_table ................................................................................................ 5ms DONE
  2020_06_28_192051_create_acl_affiliations .............................................................................................. 11ms DONE
  2020_06_29_052121_add_type_to_roles_table ............................................................................................... 4ms DONE
  2020_07_01_183306_create_acl_members_table .............................................................................................. 2ms DONE
  2020_07_30_204138_create_corporation_member_trackings_table ............................................................................. 7ms DONE
  2020_09_07_192451_add_applicationable_to_applications_table ............................................................................. 6ms DONE
  2020_09_10_192451_create_enlistments_table .............................................................................................. 2ms DONE
  2020_10_06_193916_add_status_and_comment_to_applications_table .......................................................................... 1ms FAIL

In Connection.php line 822:
                                                                                                                                                                     
  SQLSTATE[42000]: Syntax error or access violation: 1101 BLOB, TEXT, GEOMETRY or JSON column 'comment' can't have a default value (Connection: mysql, SQL: alter t  
  able `applications` add `status` enum('open', 'accepted', 'rejected') not null default 'open', add `comment` longtext not null default '', add `causer_type` varc  
  har(255) null, add `causer_id` bigint unsigned null)                                                                                                               
                                                                                                                                                                     

In Connection.php line 574:
                                                                                                                                    
  SQLSTATE[42000]: Syntax error or access violation: 1101 BLOB, TEXT, GEOMETRY or JSON column 'comment' can't have a default value  
                                                                                                                                    

```


This is my updated .env file.
```
APP_NAME=Laravel
APP_ENV=local
APP_KEY=REDACTED
APP_DEBUG=true
APP_URL=https://seatplus.test

DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=seatplus
DB_USERNAME=root
DB_PASSWORD=

BROADCAST_DRIVER=log
CACHE_DRIVER=redis
QUEUE_CONNECTION=redis
QUEUE_DRIVER=redis
SESSION_DRIVER=file
SESSION_LIFETIME=120

REDIS_CLIENT=phpredis
REDIS_HOST=127.0.0.1
REDIS_PASSWORD=null
REDIS_PORT=6379

EVE_CLIENT_ID=REDACTED
EVE_CLIENT_SECRET=REDACTED
EVE_CALLBACK_URL=https://seatplus.test/auth/eve/callback

COMPOSER_MEMORY_LIMIT=-1
```

**Reply by herpaderpaldent** [reply]
Hi

Thank you for your prompt response. I appreciate your engagement in this matter. Upon reviewing the current setup, I've identified a few areas that require attention.

First of all: Seatplus uses MariaDB@10.7 and when i looked up compatibility to MySQL Versions, that this version has run out of support. Which shouldn't concern us right now, to much, but i seek to update it at least 10.11 (LTS) but i haven't had the time to test it out, so i can't recommend it for production. 
I have experimented with switching to postgres but before i continue i want to invest some time in e2e testing (cyprus).

Not that it matter but i use php 8.2 currently and soon i will up the requirement.

When using local installation you might want to checkout [herd](https://herd.laravel.com/) and https://stackbricks.app/ (requires docker to spin up the db and redis)

Lastly: the migrations did not complete. make sure to use mariaDB to be able to use  the app. But i really would encourage you to use the docker version. There is no reason not to.

**Comment by mikeamcbrien** [comment]
Thanks for getting back with me, I do have some special circumstances and would prefer to continue to use Valet and not Docker. 😃

il’ll give mariadb a go and see if the helps me move along further and upgrade to PHP8.2. Docker is not something I want to use in production, so I want to remove it from my local environment first. 

Along my journey I’ll ensure I document and share this so others can also install without Docker.

Thanks you for looking into this and thank you for your testing and response!

**Reply by herpaderpaldent** [reply]
Fair point. You seem to be enough competent to have your own way. I simply have not enough capacity to support anyone with their custom setup and issues, that's why i ship seatplus with an environment i can control and vouch for. 

question: any opinions regarding preferred DBMS?
